src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -1040,6 +1054,246 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                 return _valueMap.GetGetter(input, ColMapNewToOld[iinfo]);
             }
 
+            public void SaveAsOnnx(OnnxContext ctx)
+            {
+                const int minimumOpSetVersion = 9;
+                ctx.CheckOpSetVersion(minimumOpSetVersion, LoaderSignature);
+                Host.CheckValue(ctx, nameof(ctx));
+
+                for (int iinfo = 0; iinfo < _parent.ColumnPairs.Length; ++iinfo)
+                {
+                    string inputColumnName = _parent.ColumnPairs[iinfo].inputColumnName;
+                    string outputColumnName = _parent.ColumnPairs[iinfo].outputColumnName;
+
+                    if (!_inputSchema.TryGetColumnIndex(inputColumnName, out int colSrc))
+                        throw Host.ExceptSchemaMismatch(nameof(_inputSchema), "input", inputColumnName);
+                    var type = _inputSchema[colSrc].Type;
+                    DataViewType colType;
+                    if (type is VectorDataViewType vectorType)
+                      colType = new VectorDataViewType((PrimitiveDataViewType)_parent.ValueColumnType, vectorType.Dimensions);
+                    else
+                      colType = _parent.ValueColumnType;
+                    string dstVariableName = ctx.AddIntermediateVariable(colType, outputColumnName);
+                    if (!ctx.ContainsColumn(inputColumnName))
+                        continue;
+
+                    if (!SaveAsOnnxCore(ctx, ctx.GetVariableName(inputColumnName), dstVariableName))
+                        ctx.RemoveColumn(inputColumnName, true);
+                }
+            }
+
+            private void CastInputToString<T>(OnnxContext ctx, out OnnxNode node, string srcVariableName, string opType, string labelEncoderOutput)
```

Comment:
I feel this code along with the other CastInputToX<> could be rewritten to be one single function, if you also add an extra parameter, such as "string toType", which accepts "string", "int", "float", and then inside the CastInputToX decide if it's going to use "keys_strings" or "keys_ints" or "keys_floats", and what "item => ConvertToX(item)" Delegate to use.